inventory_management_system_api/core/object_storage_api_client.py

```diff
@@ -0,0 +1,61 @@
+"""
+Module for providing an implementation of a client class for interacting with the Object Storage API.
+"""
+
+import requests
+from inventory_management_system_api.core.config import config
+from inventory_management_system_api.core.exceptions import ObjectStorageAPIAuthError, ObjectStorageAPIServerError
+
+
+class ObjectStorageAPIClient:
+    """
+    Client class for interacting with the Object Storage API.
+    """
+
+    @staticmethod
+    def delete_attachments(access_token: str, entity_id: str) -> None:
+        """
+        Delete attachments associated with the given entity ID.
+
+        :param access_token: The JWT access token for auth with the Object Storage API.
+        :param entity_id: The ID of the entity whose attachments should be deleted.
+        """
+        ObjectStorageAPIClient._delete("/attachments", access_token, entity_id)
+
+    @staticmethod
+    def delete_images(access_token: str, entity_id: str) -> None:
+        """
+        Delete images associated with the given entity ID.
+
+        :param access_token: The JWT access token for auth with the Object Storage API.
+        :param entity_id: The ID of the entity whose images should be deleted.
+        """
+        ObjectStorageAPIClient._delete("/images", access_token, entity_id)
+
+    @staticmethod
+    def _delete(endpoint: str, access_token: str, entity_id: str) -> None:
+        """
+        Sends a `DELETE` request to the Object Storage API with the provided JWT access token as a header and the entity
+        ID as a query parameter.
+
+        :param endpoint: The Object Storage API endpoint to send the request to.
+        :param access_token: The JWT access token for auth with the Object Storage API.
+        :param entity_id: The ID of the entity whose objects should be deleted.
+        :raises ObjectStorageAPIAuthenticationError: If auth with the Object Storage API fails.
+        :raises ObjectStorageAPIServerError: If any other error is encountered while communicating with the Object
+            Storage API.
+        """
+        url = f"{config.object_storage.api_url}{endpoint}"
+        headers = {"Authorization": f"Bearer {access_token}"}
```

Comment:
Should this logic be optional for when auth is disabled?

Reply:
The thing is that theoretically auth can be disabled in the IMS API but enabled in the Object Storage API so requests will error. If we are okay with this then I can change it to be optional.

Reply:
I would personally think if you are using it with authentication, they should both be with authentication. If we only needed to do something for a particular kind of test I would think its fine to just manually comment out the line kind of like we do testing extremely short refresh tokens/checking front end auth by returning specific responses.

Reply:
I addressed this, cheers. What about timeout for the request itself? Pylint complains about it by default hence why I put `# pylint: disable=missing-timeout`.

Reply:
10 seconds would be more than enough I would think, pylint just likes it to have something so doesn't hang forever, so as long as its longer than we would ever expect it to take its fine. If we expect it will take longer if there's more to delete then maybe even 60 to be safe.